framework/src/modules/chain/bft/finality_manager.js

```diff
@@ -248,20 +253,16 @@ class FinalityManager extends EventEmitter {
 				) {
 					return needleHeight;
 				}
-
 				// Move the needle to previous block and consider it current for next iteration
-				needleHeight = previousBlockHeader.maxHeightPreviouslyForged;
```

Comment:
This line was doing the jump to previous block header of that delegate. That's a required behaviour as we don't need to iterate linearly for each block header.

Reply:
Does it work in case of conflicting block header?

Reply:
As far as block at `previousBlockHeader.maxHeightPreviouslyForged` is valid block forged by same delegate, it should work.